Tango skin :: update

ronso0 opened the pull request:
In this PR I fix some minor issues and clean up a bit.

- [x] **1**) identical beatsize spinboxes for left and right decks
- [x] **2**) identical EffecttSelector for left and right units
- [x] **3**) identical Loop sections for left and right decks
- [x] **4**) visualize flow within effect chains
- [x] **5**) when FX lot is focused, FX focus button now can un-focus with left-click as well
- [x] **6**) redesign FX focus button so that it's not mistaken for an On/Off button for the effect
- [x] **7**) optional SuperKnob
- [x] **8**) assign buttons for Master/Headphone in collapsed FX units
- [x] **9**) simplify channel mixer, make Pfl button static
- [x] **10**) indicate active passthrough while Vinyl Controls are hidden
- [x] **11**) implement EffectPushButton
- [x] **12**) ~~hide headphone mixer when no headphone device is configured~~
- [x] **13**) add booth volume control if booth output is configured
- [x] **14**) make all skin settings fit on minimal screen
- [x]  **15**) grey out headphone controls if no respective device is set up
- [x]  **16**) Pfl & xfader assing buttons for ~~Mic~~/Aux
- [x]  **17**) check Library colors
- [x]  **18**) make all mixerbar gain controls knobs
- [x]  **19**) add larger Master level meter

For booth gain tooltip, #1343 needs to be merged.

Please test if this makes sense. Also please check for glitches in Effect Selector drop-down and beat-jump/beat-loop size boxes, especially on Windows & OSX

Commit:
fix skin settings' Spinny/Cover options

Commit:
make EffectSelector identical in left & right FX units

Commit:
push FX units outwards to compensate FX label

Commit:
revert symmetrical beatsize spinboxes, add size-aware fillers

Commit:
rename/move cover graphics

Commit:
Library: brighter font color for selected line

Commit:
clean up knob_textless.xml

Commit:
FX: visualize FX chain flow

Commit:
Skin Settings: typo

Commit:
Merge branch 'master' of http://github.com/mixxxdj/mixxx into Tango-update

Be-ing commented:
Hmm, should the metaknobs for effect units on the right be moved to the left side of the comboboxes to mirror the other side?

>  4) visualize flow within effect chains

I like this.

I made some changes to the EQ knobs and effect parameter knobs in #1315, so if you haven't already, merge the latest updates from master into this branch to make sure you don't make merge conflicts.

Commit:
FX focus button: when active, add 'unfocus' as left-click action

Commit:
FX: restructure/clean up

Commit:
Deck buttons & Hotcues clean-up

Commit:
General/FX clean-up

ronso0 commented:
> Hmm, should the metaknobs for effect units on the right be moved to the left side of the comboboxes to mirror the other side?

In collapsed FX units I made toggles/meta/focus identical for both sides because IMO it resembles the left-to-right order in FX units on controllers, so I think symmetry is not necessary then.
However, when expanded layout changes completely, so I think it makes sense to have it symmetrical then.

ronso0 commented:
**5**) when FX lot is focused, FX focus button now can un-focus with left-click as well
It just felt strange to have a button that doesn't react to left-click in one state
**6**) redesigned FX focus button so that it's not mistaken for an On/Off button for the effect

Commit:
restore blank default button backpath

Commit:
Record: add Rec text/icon when active

Commit:
FX: style icons

Commit:
FX: add optional Super Knob

Commit:
rename cf > xfader

Commit:
FX: redesign/add Head/Master assign buttons to collapsed unit

Commit:
Mixer clean-up

Commit:
clean-up

Commit:
finish Play button

Commit:
fix Crossfader/Skincontrols in netbook mode

Commit:
Decks: simplify code

Commit:
clean up code

Commit:
Mixer: simplify EQ knob templates

Commit:
FX Selector: finish drop-down button

Commit:
Decks: add hint for active passthrough while Vinyl Controls are hidden

Commit:
clean up vector images

Commit:
clean up code

Commit:
Decks: fix Time/Title/Artist section

Commit:
clean-up/rename

Commit:
Decks/Skincontrols: adjust spacing

Commit:
fix for fx_units_spacer.xml

nikmartin commented:
I just wanted to comment to say I've been using Tango full time since the new re-loop features were merged to master and tango implemented them, and this skin has been awesome. Everything is well laid out, and after spending some time practicing, everything is in the right place. It took me a while to figure out how to assign effects to my headphones/cue so i could test them before sending them to main, but overall I'ts pretty intuitive. There are some usability issues with the library, but I think those are not specific to the skin. I will triage those and add bugs to launchpad later. It involves when scrolling folders the file list will not update until the folder is clicked twice with the mouse, so the selected folder will have the wrong file list until I click it again. Using the DDJ-SBJ mapping, using the encoder knob to select it works fine.

nikmartin commented:
I just merged this pr into master and rebuilt, and the samplers disappear in the UI. They still play, but are not displayed. The SAM on/off button works as if they are being displayed, but they don't. Going to the menu and clicking [1]Samplers does nothing, doesnt increment the count, etc

Commit:
re-enable samplers

ronso0 commented:
Thanks for testing!

> I just merged this pr into master and rebuilt, and the samplers disappear in the UI. They still play, but are not displayed. The SAM on/off button works as if they are being displayed, but they don't. 

Whoopsy, my mistake: I commented them out in skin.xml to accelerate skin reloading. Just fixed it.

> Going to the menu and clicking [1]Samplers does nothing, doesnt increment the count, etc

Even with Samplers available, this button apparently only works if Samplers are visible.
Don't know why..

ronso0 commented:
Updated screen shots in first post

nikmartin commented:
and i just went a did a git reset hard back to master, rebuilt, and they showed up

nikmartin commented:
OK, merged the pr back in - do I actually need to rebuild with scons for skin updates?

ronso0 commented:
> do I actually need to rebuild with scons for skin updates?

Actuallly you don't. You can either start mixxx with "--ressourcePath /path/to/res/folder", or you copy/paste the skin to your install directory, or you put a link there like "Tango-update".
Same applies if you like to test controller mappings etc.

Commit:
Implement EffectPushButton for Fx Units and EQ Kill buttons

ronso0 commented:
Just finished implementing EffectPushButton for Fx Units and EQ kill buttons.
For respective tooltips to show up, #1332 needs to be merged.

I'd say I'm done here, please let me know if you discover any bugs.
Should the QuickEffect knob be optional?

Be-ing commented:
>  Should the QuickEffect knob be optional?

The QuickEffect knob should automatically hide when no QuickEffect is loaded.

I'll test this soon.

Commit:
spacers in EQ & QuickEffect knob, FX variables

Commit:
clean up button templates

Commit:
fix resolved stash/merge conflicts

Commit:
surprise

Be-ing commented:
* Permanent deck controls in options menu do not fit in minimum height.
* Highlighted library row colors are bit hard to read. Maybe make the background highlighting a bit darker?
* Good job figuring out a way to get left clicking to unfocus effects! I'll implement that in Deere too.
* I think the "X" icon used for unfocusing could get confused with disabling the effect.
* Toggling maximized library is slow.
* Effect selector dropdown looks slightly offset to the right (on GNU/Linux, Qt4):
![effect-selector](https://user-images.githubusercontent.com/9455094/29996446-18b11624-8fc4-11e7-9bae-526daf4398ef.png)

Be-ing commented:
Overall I like the changes.

* The mixing of symmetry and asymmetry with the loop controls is a little weird thinking about it, but somehow it feels like it makes sense.
* Permanent deck controls in options menu do not fit in minimum height.
* Highlighted library row colors are bit hard to read. Maybe make the background highlighting a bit darker?
* Good job figuring out a way to get left clicking to unfocus effects! I'll implement that in Deere too.
* I think the "X" icon used for unfocusing could get confused with disabling the effect.
* Effect selector dropdown looks slightly offset to the right (on GNU/Linux, Qt4):
![effect-selector](https://user-images.githubusercontent.com/9455094/29996446-18b11624-8fc4-11e7-9bae-526daf4398ef.png)
* Toggling maximized library is slow.

Be-ing commented:
Could you add a booth gain control to the mixer bar that only shows when [Master],booth_enabled is 1?

ronso0 commented:
> The mixing of symmetry and asymmetry with the loop controls is a little weird thinking about it, but somehow it feels like it makes sense.

I had some difficulties with the former layout where the loop sections were mirrored. This feels better and I hope this works for others as well.

> Permanent deck controls in options menu do not fit in minimum height.

It appears that only the buttons in "Decks/General" shrink to less than 15px. I'll find a way

> Toggling maximized library is slow
With master or library redesign branch?

Be-ing commented:
>> Toggling maximized library is slow
>With master or library redesign branch?

With master. I'm don't know if there is a good way around this slowness of hiding/showing QWidgets.

Commit:
Merge remote-tracking branch 'origin/master' into Tango-update

ronso0 commented:
Okay.
Here it takes about as long as switching 2/4 decks, or toggling samplers, no longer than halve a second.
If this is a problem, I don't know how to solve it from the skin side.

ronso0 commented:
> Could you add a booth gain control to the mixer bar that only shows when [Master],booth_enabled is 1? Also, you can hide the headphone controls when [Master],headEnabled is 0.

Sure.

> we should hide the PFL buttons when headEnabled is 0 as well

I'd rather grey them out if no headphone device is configured.
Or -even better IMO- pop up a message "No headphone device configured" with a button to go to Sound Preferences directly, like it happens with vinyl passthrough with no dvs device set up.

ronso0 commented:
> Effect selector dropdown looks slightly offset to the right (on GNU/Linux, Qt4)

Interesting, how it looks on your system. Which desktop do you use?
What exactly bothers you: that the list's left edge is aligned to WEffectSelector button?
On xfce it looks like this
![tango-effect-dropdown](https://user-images.githubusercontent.com/5934199/30249618-e1444e22-963f-11e7-9ba4-23ec8941ebf2.png)

Anyway, I tried a few more things to style the list, but to no avail, it would break other properties.

Commit:
styling WEffectSelector & WBeatSpinbox

Commit:
hide mixerbar Headphone section when no device is set up

Be-ing commented:
> Interesting, how it looks on your system, especially the border radius and that the list is positioned below the button. Also I never manged to make scrollbars appear.
Which desktop do you use?

Yeah, that is interesting that it looks different. I use KDE with the Oxygen widget theme and Obsidian Coast color theme.

>  And what exactly bothers you: that the list's left edge is aligned to WEffectSelector button?

It looks a little weird that the scroll bar is outside the width of the unexpanded widget, but it's not a big deal.

> Anyway, I tried a few more things to style the list, but to no avail, it would break other properties.
I.e., there's no way to add the padding to the items themselves, only to the list.

IIRC I had the same issue when working on Deere and didn't figure out any way to resolve it. I think it's good enough to leave it as it is.

ronso0 commented:
> IIRC I had the same issue when working on Deere and didn't figure out any way to resolve it. I think it's good enough to leave it as it is.

Yes, I already gave up on this.. ;)

ronso0 commented:
> I think the "X" icon used for unfocusing could get confused with disabling the effect

Am I right that Focus buttons only appear if a controller mapping specifically enables them?
I notice that my ToDo **6)** is kind only relevant when a completely new mixxx user mixes controller and mouse. Other than that, the new style only helps to distinguish Tango focus buttons from Deere effect enable buttons.

Commit:
FX units fine tuning

Commit:
Mixerbar: add xFader & Pfl buttons, rework colors

Commit:
qss clean-up

Be-ing commented:
> Pfl & xfader assing buttons for Mic/Aux

Mics actually can't be assigned to the crossfader in the engine.

Be-ing commented:
Why is there a toggle button for the library? What is a use case for hiding it?

ronso0 commented:
> Why is there a toggle button for the library? What is a use case for hiding it?

1. Same use case that justifies hiding artist/title in Deere: not everybody needs to know everything
2. Having a clean UI when using tracks from external sources (drag'n'drop) or passthrough

Commit:
Decks: add hint that passthrough is active

Commit:
Mixerbar fixes

Commit:
grey out Headphone controls if no respective device is configured

Commit:
Mixerbar: add Booth gain, visible if booth device configured

Commit:
Library: improve colors

ronso0 commented:
IMO I'm done here, please test.

@Be-ing Do the library colors look okay on your screen now?

Commit:
Skin Settings fit now

ronso0 commented:
Now I'm done.

Skin Settings fit, and I reworked the indicator for when the channel mixer displaces the big spinny or cover art. IMO it is absolutely clear now what's happening if you watch what you're clicking.

Be-ing commented:
* Booth gain shows the master gain tooltip. Set the TooltipId to "booth_gain", which is what I am using in #1343 (note this tooltip doesn't exist yet).
* Clear text icon for search bar is misaligned:
![image](https://user-images.githubusercontent.com/9455094/30767899-e291e8ee-9fc6-11e7-884f-1c0123421668.png)
* Effect unit expansion button shifts slightly when hovered. When the effect unit is collapsed, it shifts down but when it's expanded the button shifts up.
* The "Permanent VU meter" option is oddly named. It seems this shows the meters when the mixer is hidden? Also, [they're not really VU meters](https://bugs.launchpad.net/mixxx/+bug/1429760). IMO "level meters" is the best term for what we have.
* "Super Knob (for Meta Knobs)" is also oddly named. I think it should just say "Super Knob" in the skin settings menu. What the super knob does is explained by its tooltip.
* Could you remove the effects assignment switches from the samplers and the toggle for them from the skin settings menu? Letting samplers be assigned to effects requires a ridiculous amount of memory to be allocated when loading effects. Refer to [this comment](https://github.com/mixxxdj/mixxx/pull/1254#issuecomment-312782912) for details.
* I still think there isn't enough color contrast between the text and background color for the highlighted row in the library:
![image](https://user-images.githubusercontent.com/9455094/30768173-af223308-9fc9-11e7-92b2-475db5618697.png)

ronso0 commented:
Thanks for your review!
I'll take care of the issues.

> Clear text icon for search bar is misaligned:

A noticed long ago, how to fix it? What's the internal name of that button?

> Effect unit expansion button shifts slightly when hovered. When the effect unit is collapsed, it shifts down but when it's expanded the button shifts up.

Consider it a feature. It's intended and consistent for all minimizeable containers like Vinyl Controls, Deck Controls, Samplers etc., indicating what will happen after button press.

> I still think there isn't enough color contrast between the text and background color for the highlighted row in the library:

okay, a screenshot doesn't help much, I'll have to check this on another machine myself.

> Could you remove the effects assignment switches from the samplers and the toggle for them from the skin settings menu? Letting samplers be assigned to effects requires a ridiculous amount of memory to be allocated when loading effects. Refer to this comment for details.

Hm.. There would be bug reports complaining about that missing feature I guess.
but I see the memory introduced in #1254 
Am I correct that effects would allocate memory only if **a**) a sampler is assigned to them and **b**) that sampler is playing?
If not, could that be a solution to implement in #1254 ?

Be-ing commented:
> okay, a screenshot doesn't help much, I'll have to check this on another machine myself.

To clarify, I can read the text currently, but I think it could be easier to read with different colors.

> Hm.. There would be bug reports complaining about that missing feature I guess.

Although assigning samplers to effects was available in the engine in Mixxx 2.0, it was not exposed in any skins so I don't think anyone will miss it. It is impractical to manage effect assignments individually for lots of samplers. AFAIK no one has asked for this feature. I only added it to Deere in #940 because I could. Now I understand it has performance consequences though. Removing 4 buttons from the samplers should speed up skin load/shutdown time as well.

> Am I correct that effects would allocate memory only if a) a source is assigned to them and b) that source is playing? If not, could that be a solution to implement in #1254 ?

No, that is not correct. Effects allocate memory in the main thread for every channel they can be assigned to when they are loaded, then the effect object is made available to the engine thread on the next audio callback. Dynamically allocating memory for effects when they are assigned to a channel would require memory allocation in the engine thread, which would likely cause xruns.

ronso0 commented:
> No, that is not correct. Effects allocate memory in the main thread for every channel they can be assigned to when they are loaded

It's a pity. I'll remove those buttons, no problem.

Re: loading time
I always was under the impression that waveform overview and svg buttons increase loading time, not a simple text pushbutton

Be-ing commented:
Every widget in the sampler template is created 64 times, so every one makes a difference for the load/shutdown time.

Commit:
Samplers: remove FX assign buttons

Commit:
fix skinsettings and booth tooltip

Commit:
Decks: block Play button when passthrough is active

Commit:
qss/xml clean-up & simplification

ronso0 commented:
I took care of the issues @Be-ing discovered.
Anyone else tested this? Found any issues?

If not, this is ready to merge.

nikmartin commented:
I will test this over the next few days.

Commit:
minor fixes & renaming

Be-ing commented:
I noticed there is no DSP/audio buffer usage meter ([Master], audio_latency_usage Control).

>  12) hide headphone mixer when no headphone device is configured

I see this grays out the controls in the mixer bar. I think it should hide them. I don't think there's a point to cluttering the UI with unusable controls. Leaving the PFL buttons on screen but grayed out makes sense I think because it gives a hint that there is functionality that is unavailable. But without a headphone output, I don't think there's any need to hint that there is a headphone gain, head mix, or split cue control.

I think a little more horizontal padding between the sections of the mixer bar would be helpful.

Regardless of the above, I think there's enough improvements in this branch already for it to be merged as-is. There will always be room for improvement so let's not wait forever for perfection.

ronso0 commented:
> I noticed there is no DSP/audio buffer usage meter ([Master], audio_latency_usage Control).

I'll have a look where I can implement that. I think I'll use different colors than those of the volume level meters.

> I see this grays out the controls in the mixer bar. I think it should hide them. I don't think there's a point to cluttering the UI with unusable controls...

You can collapse headphone section, like Mic & Aux, if you don't need it.

I'd like to get this merged, too, and am waiting for reviews. Until then I'll commit fixes when I stumble across small issues.

nopeppermint commented:
I just tested this Skin Update with mixxx-2.1.0-alpha-pre-master-git6340-release-xenial-amd64.deb from 10-Sep-2017 12:52 under Linux Mint 18.1 (Cinnamon)
I just found the following Issue:
orange (and grey) preview headphone icons are placed to low
![bildschirmfoto vom 2017-09-26 23-18-42](https://user-images.githubusercontent.com/3403218/30884817-5314f460-a311-11e7-9321-0cdf86ca8748.png)

nopeppermint commented:
I am wondering if there should be an arrow up and down before "sec" ?
![bildschirmfoto vom 2017-09-26 23-26-44](https://user-images.githubusercontent.com/3403218/30885148-8375fd7e-a312-11e7-8ac5-599c8294aba1.png)

But to be honest this "Error" exist in all Skins...

daschuer commented:
Where is the right play/pause button?

daschuer commented:
Compact mode is less compact than normal mode.

nopeppermint commented:
in Tango Skin from mixxx-2.1.0-alpha-pre-master-git6340-release-xenial-amd64.deb 
the option "VU Meter" under Mixer had an action (show/hide VU Meter)

In the new Tango Update Skin it's renamed to "Permanent Level Meter" and does not have an action (Level Meters are displayed always).

nopeppermint commented:
> Compact mode is less compact than normal mode.

plus on small screens you can not switch from "compact" to "normal mode".
There is a "Compact mode" you can select/deselect but there is no change of "mode" if the window is to small!

Commit:
fix Play button and Fx button label

Commit:
re-enable samplers

ronso0 commented:
> orange (and grey) preview headphone icons are placed to low

Afaik I didn't change that, will have a look.

> Where is the right play/pause button?

Fixed. Forgot to workaround the [stacked layout bug](https://bugs.launchpad.net/mixxx/+bug/1627859) there.

ronso0 commented:
'Compact Mode' was introduced in the previous update. Since then Tango requires less screen width which makes it fit netbook screens, simply by moving the central button/info section to the top in form of a toolbar. I renamed it because it's more compact in horizontal dimension.

On small screens it switches automatically, instead of saying "You can't use this skin because your screen is too small" or squeezing skin elements.
How do you think the irritation you experienced could be avoided?
Rename this feature back to Netbook Mode?
Grey out the skin settings button if Tango is in this mode automatically?

ronso0 commented:
> In the new Tango Update Skin it's renamed to "Permanent Level Meter" and does not have an action (Level Meters are displayed always).

I renamed it after @Be-ing pointed me to the respective bug about mixxx VUs are no 'real' VUs but non-standard level meters.
I decided to have them always visible in Tango's channel mixer because they are permanent in other skins, as well as on pretty much any hardware mixer.
If you hide the channel mixer while mixxxing with a controller, you can show standalone level meters if your controller doesn't have them.

Be-ing commented:
> If you hide the channel mixer while mixxxing with a controller, you can show standalone level meters if your controller doesn't have them. So they are 'permanent' even with hidden channel mixer.

Why is this an option? Why not always show them?

Commit:
Library: fix Preview button position

ronso0 commented:
> Why is this an option? Why not always show them?

Please remember that I started Tango in forums as an controller extension, meaning all (most) redundant interface elements already supplied by users' controllers can be removed from screen.
For example if you have a full featured controller, you'd need the screen only for artist/title, effect management, samplers and the library, of course, but no EQs, Vol faders, crossfader etc

nikmartin commented:
> meaning all (most) redundant interface elements already supplied by users' controllers can be removed from screen.

I am a HUGE supporter of this and use the Tango skin exclusively. If it is easily accessible on my controller, I don't need it in the UI. My DDJ SB2 doesn't have spinnies, so I turn those on, and it doesn't have a clock, so that's important to me. I do miss the latency meter, it could even be 4 pixels wide across the top of the center cluster, but knowing if I'm beating up my CPU and possibly getting ready to glitch audio is pretty important.

ronso0 commented:
Thanks @nikmartin 

I'll put the latency meter below the clock, 4px tall, but I was facing some issues.
Stay tuned

Be-ing commented:
> For example if you have a full featured controller

Ah, I forgot that many controllers have meters on them because mine does not.

daschuer commented:
> Grey out the skin settings button if Tango is in this mode automatically?

Yes! ... or remove the button entirely.

Commit:
add audio latency meter

ronso0 commented:
> Ah, I forgot that many controllers have meters on them because mine does not.

Than this feature is for users like you -_o

ronso0 commented:
>  >  Grey out the skin settings button if Tango is in this mode automatically?

> Yes! ... or remove the button entirely.

I know there was a reason to introduce this button, but I can't recall right now.
Will see what I can do...

Commit:
skin settings polish

ronso0 commented:
Compact Mode is now Netbook Mode again, hope it's clearer. That button is greyed out if skin switched to netbook mode automatically due to small screen.
Ploished a few other settings as well, I just hope it looks with all screen sizes now.

Put audio latency usage is below clock.

ronso0 commented:
Off topic:
When mixxxing with mouse and keyboard only, I miss the fixed-size loop grid and think about re-implementing it. For example, setting up a 1-beat-loop takes too long with spinboxes and loop button.
What do you think?

Be-ing commented:
I'm not in favor of bringing back the clutter of the whole fixed loop size button grid, but maybe a limited set of buttons (1 - 16 beats?) could be shown with a skin option.

ronso0 commented:
Yeah, I know the grid was removed for a reason, but it's really helpful in some situations.
I imagine to have an expand/collapsed toggle in the looping section, like the one for deck controls.

Commit:
small fix in skin settings

Commit:
update skin version, small size fix

Commit:
Decks: hide controls toggle if unnecessary

Commit:
adapt netbook toolbar to resized skin elemnts

Commit:
update default skin settings

Commit:
fix Aux control and tooltip

Commit:
..and fix Aux mute icon

InannaMoon commented:
One more minor issue I found: In the AUX inputs, the tip balloon that pops up says "Microphone Talk-Over". Otherwise, the icon is fine and the button operation is fine.

Commit:
orange border when active + hover

Commit:
fix FX units focus appearance

Commit:
add nicer mask for big spinny

Commit:
make Hotcue skin settings button consitent

Commit:
fix & style beat spinbox

Commit:
improve FX focus UX

Commit:
ressurect fixed loopsizes, adapt mini/Maxi button

Commit:
fix FX flow graphics

Commit:
fix tooltips, polish buttons

Be-ing commented:
There's still a toggle for the Mic/Aux effects routing switches in the skin settings menu although the switches have been removed from the skin.

ronso0 commented:
Sure, both FX assign buttons for Mic/Aux and respective toggle in skin settings are still there.
As you suggested, I removed those only from _Samplers_ due the memory allocation issue with a lot of samplers.

Commit:
tiny fix for mute/talkover button

InannaMoon commented:
I got a zip of your repo, again, and copied out the Tango skin with undated files dated 16 Oct 2017. Same behavior for the AUX input mute/unmute button as in my previous comment.

ronso0 commented:
> I got a zip of your repo, again, and copied out the Tango skin with undated files dated 16 Oct 2017. Same behavior for the AUX input mute/unmute button as in my previous comment.

I changed the tooltips in the [previous commit](https://github.com/mixxxdj/mixxx/pull/1330/commits/e1c0a461db625edca92ab730aa112d3e6a3a4ee4#diff-dcadcd74bc6bacd4657ca9571f94cd66R28), something has gone worng on your side. Maybe you didn't remove the previous skin version?

Commit:
update skin version & description, clean up

ronso0 commented:
I'm done with polishing, just updated the skin version and the description.
Ready to merge

InannaMoon commented:
I know it will be a tight fit, but I request headphone sends for both MIC and AUX inputs. (Maybe change the FX1 - FX4 sends to F1 - F4.) (LateNight has headphone sends for MICs and AUXs.) Thanks.

ronso0 commented:
Indeed it gets tight but it looks okay since I introduced it in 8dde175d795eaf79762e43ee7b1f598de19922f7
I just double-checked, if you pull the branch it looks like this:
![tango-update-525edca](https://user-images.githubusercontent.com/5934199/31819672-fbd2f770-b59d-11e7-81db-69efc569bd91.png)

How many Aux and Mics do you usually use?

Be-ing commented:
Where is the master level meter?

InannaMoon commented:
Thanks. Looks nice to me.

How many mics and aux I use depends.

For a solo, on-line stream-cast, 1 mic and 2 aux. The 2 aux are connected to virtual audio devices for VOIP and Discord.

With an in-studio guest, 2 mics and 1 or 2 aux.

When I "go mobile", usually just 2 mics, sometimes also an aux.

Commit:
prevent long track comments from stretching decks

ronso0 commented:
> Where is the master level meter?

![tango-update-d9280e4](https://user-images.githubusercontent.com/5934199/31853835-dc744e90-b68f-11e7-9b30-69d2412aeb4d.png)

InannaMoon commented:
I successfully loaded the updated Tango. The AUX mute button tip is now correct and I get the headphone sends for MIC and AUX.

One small item: When the "netbook" option is selected, the clock widget isn't wise enough to show all the digits:
[![Mclock.png](https://s19.postimg.org/9pg8tqgyr/Mclock.png)](https://postimg.org/image/h5fifj4nz/)

Commit:
implement control vinyl status light

Be-ing commented:
How about moving the master level meter to the mixer and the CPU usage meter to the toolbar?

ronso0 commented:
> When the "netbook" option is selected, the clock widget isn't wide enough to show all the digits

Strange, I explicitely disabled seconds in the clock. I guess you run Windows?

ronso0 commented:
> How about moving the master level meter to the mixer and the CPU usage meter to the toolbar?

I'd keep the CPU usage meter where it is so it's visible even if controller users don't need the mixerbar.
I could put the master level meter in a separate column next to the GUI controls, but I'm not really happy with the asymmetry.

Commit:
add tooltip for latency usage meter

Commit:
test: master level meter

Commit:
lateny tooltip for netbook mode

ronso0 commented:
> What's the level meter under the clock?

That shows latency used for audio, I forgot to add the tooltips. Done

ronso0 commented:
> FYI, I dual boot Linux (Dragora) and Win7 Pro. I have Mixxx installed in both. Currently, I've been booting to Win7 Pro to run a work related application.

Okay, is the time displayed without seconds if you run mixxx in Linux?

ronso0 commented:
@Be-ing I implemented an additional larger Master level meter in between channel mixers.
Along with the channel level meters it can be shown permanently.

Commit:
repair Master VU whoopsy

Commit:
adapt responive spacers

InannaMoon commented:
Yes, in Linux time displays without seconds.

I tried both with and without netbook setting. In Win7, the M in the AM/PM was cutoff in non-netbook mode.

For netbook mode, the clock widget could easily be wider, though not in non-netbook mode.

Commit:
enhance floating level meters

Commit:
enhance crossfader scale

Commit:
knob redesign proposal n°1

ronso0 commented:
The last commit is a proposal to re-think the knobs.
Both knob pointer and background should, but indicate the end positions. Markers (either dots or strokes) somehow don't fit Tango style, so I tried rings cropped at the end positions.
IMO emphasizing knobs' centers would only make sense for EQ knob since it matches the default position.

Please test.

Before
![tango-update-knobs-old_crop](https://user-images.githubusercontent.com/5934199/31896689-ac81ae36-b814-11e7-8b4d-277b52e3858f.png)

Now
![tango-update-knobs-new3_crop](https://user-images.githubusercontent.com/5934199/31896696-b21ffa6e-b814-11e7-9204-54c37f105901.png)

Commit:
logo, spacer fixes

Commit:
TEST remove hidden waveforms

Be-ing commented:
I like the depiction of the ranges with the knobs' rings.

The master level meters don't have tooltips.

naught101 commented:
Moving from https://github.com/mixxxdj/mixxx/pull/1357#issuecomment-338531602

>  There are markers on the Vol fader and center marks on the pitch fader. How do you imagine minimal position markers for the knobs?

Yeah, true, the Tango faders do have markers, but the tempo faders only have centre markers, and the colours are inconsistent between the deck volume (black), and tempo and cross faders (light grey). I also don't love the line intersections - the circles seem inconsistent with the rest of the style. And the actual faders look too thin to me. Also, the 0% and 100% markers are cut in half on each fader, making them look thinner.

Maybe something like this would work:

![tango fader](https://user-images.githubusercontent.com/167164/31869511-3f672756-b7f4-11e7-9cd4-b070f3e3642b.png)

(only using the top half for shorter faders).

Re: Tango knobs, it would be good if they had 0, 50% and 100% markers, and maybe smaller 25% and 75% markers. Could just be dots, or short lines (e.g. 3-4px), to match the faders.

Commit:
refactor skin.xml

So far decks and respective centerbox were instantiated as Singletons
which then appeared depending on the amount of FX units visible.
This caused some flickering when switching between 2 and 4 FX units.
Now decks appear only once and FX units are packed into Singletons.

Commit:
TEST improve Volume fader and xfader scales

InannaMoon commented:
I like the cropped circles for the knobs. Maybe a small gap in the circle at the 50% point? Just a thought. I have no trouble without it.

Commit:
SVG optimize: resize and strip unnecessary tags with svgo

Commit:
fix Tooltip typos for level meters

ronso0 commented:
While scrolling through the dev mailing list, I came across @esbrandt hints on svg optimization.
I resized knobs to their actual render size and let [svgo](https://github.com/svg/svgo) strip all unnecessary tags. Size went down from 1.3MB to ~200kB...
Hopefully it accelerates startup on slower machines.

ronso0 commented:
> Maybe a small gap in the circle at the 50% point?

I made some variations beforehand, with dots, with gaps. The one with a center gap made the rings look kind of fragmented, like spread pizza pieces. Please send me PRs if you manage to create good-looking markers.

I made the fader notches consistent. But I don't see reasons for end markers on the pitch sliders since there's no "End", not even a lower end since the largest rate range one can adjust is 90%.
It would be nice though to have 

Re: fader colors
the pitch faders are brighter to keep a reasonable contrast on skin background.

Re: small fader handles
Consider them just indicators. Basically, one doesn't have to hit the handle: either with mouse or touch you can pick and drag any place on the fader scale.

InannaMoon commented:
The cropped circles are fine (for me) with no other marks. Was just throwing out an idea.

I would have no problem with smaller fader handles, if you want to do that. Don't know about other people,

2 things I did notice when the window is at minimum width: 1. The new master level meter "drifts" to the left. 2. The right deck gets cut off on the right. (screen picture attached).

Looking great. I like this skin a lot.

![deckr](https://user-images.githubusercontent.com/32589413/31958070-98bb2526-b8be-11e7-874f-a459cfb566c5.png)

ronso0 commented:
> The cropped circles are fine (for me) with no other marks. Was just tossing in an idea.
I would have no problem with smaller fader handles

I'd welcome markers if anyone has an idea.. I'd say the handles are fine, I argued against making them larger like in Deere

> 1. The new master level meter "drifts" to the left. 2. The right deck gets cut off on the right. (screen picture attached).

Thanks for letting me know. Became a bit lazy testing all configurations it seems

Commit:
fix squeezed decks at minimal window size

naught101 commented:
> I'd say the handles are fine, I argued against making them larger like in Deere

I don't know if it's the size, so much as the aspect ratio that bugs me. I wonder if they might work the same thickness, but a bit shorter? The red is already fine at grabbing attention.

re: markers: I think the trimmed circles are a big improvement.. Centre is less necessary, because it's easy to see when the dial is centred anyway.

InannaMoon commented:
Trying the e913a02 commit. So far looks great.

As for the knobs, I'm fine with just the cropped circles. Depending on what I'm adjusting, I'm either referencing a level meter or listening for desired result.

Thanks

Commit:
make Vol slider handles as wide as pitch handles

ronso0 commented:
commit n°99 :wine_glass:

ronso0 commented:
I reduced the Vol faders' width so it's as wide as pitch fader handle.

This PR has grown incredibly, I'll stop it now (I bet I said that before..) and fix bugs only.
Please test.

nikmartin commented:
Should there be concern for the skin errors in the terminal on startup? I'm in the 'all errors are bad errors' camp:

```
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "IN" , 0 
Warning [Main]: QString::arg: Argument missing: "OUT" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: WaveformRenderMark::setup - redefinition of "cue_point" 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: QString::arg: Argument missing: "C" , 0 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
Warning [Main]: src/widget/wpushbutton.cpp:196 SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9 <PushButton>: WPushButton::setup: Connecting a Pushbutton not in PUSH or TRIGGER mode is not implemented
 Please set <RightClickIsPushButton>true</RightClickIsPushButton> 
```

ronso0 commented:
Thanks for testing!
Afaik all those `Warning [Main]: QString::arg: Argument missing:` lines result from empty decks/samplers, nothing to worry about.

That line about `SKIN ERROR at skin:skin_settings_button_sampler_rows.xml:9` can also be ignored,  it's caused by the way the button to set the number of sampler rows is constructed. If I'd put `<RightClickIsPushButton>true</RightClickIsPushButton>` into the button template as suggested, users couldn't decrease the row number with right-click anymore, instead would have to cycle through all possibilities with left-click i.e. when going from 2 rows back to 1 row of samplers.

Any other issues?

nikmartin commented:
I'll let you know tomorrow. I have two gigs today, a wedding and a Halloween :ghost: gig at a bar later, so it will get a good workout today.

I have a confession: I was considering switching to Rekordbox because of tighter Pioneer controller integration, so I paid for a month-to-month subscription, and after 2 weeks, can't stand it and went back to mixxx+Tango skin. Mixxx has some issues that I'd like to help solve if I had more c++ + QT experience, but I'm a pure web dev (Javascript) with a little bit of embedded C experience.

Be-ing commented:
We could always use more help! I think you'd be able to pick up C++ relatively easily with that background and the help of an IDE. We have guides for setting up [KDevelop](https://mixxx.org/wiki/doku.php/kdevelop) and [Eclipse](https://mixxx.org/wiki/doku.php/eclipse) to work on Mixxx. Personally I like KDevelop.

sblaisot commented:
black on black on effects selection is hard to read:
![image](https://user-images.githubusercontent.com/2768193/32164116-c7ed3254-bd5e-11e7-8922-776a479d030d.png)

ronso0 commented:
> black on black on effects selection is hard to read

That's strange, actually a light grey is defined for the font. I guess you use Windows!?
With Linux, font color of `WEffectSelector` is used for `WEffectSelector QAbstractItemView` (drop-down list items) as well. If I define any other color explicitely for `WEffectSelector QAbstractItemView` I don't see any change (on Linux).
As a quick test you could uncomment `color: #cccccc` in style.qss (line 1336)

sblaisot commented:
Yes, this is on windows.
I uncommented `color: #cccccc` (it was on line 1331) and it's much better:
![image](https://user-images.githubusercontent.com/2768193/32183993-2ffe9676-bd9b-11e7-9d30-04789ad1495f.png)

sblaisot commented:
maybe you could make the drop-down menu a bit larger than the filter name square to allow seeing more characters of the effect name?

Commit:
EffectSelector: explicitely define fon color for items in drop-down menu

Commit:
put Master VU in a nice box

Commit:
add hover effect to play position display

Commit:
Effect Selector: re-enable tick mark for active effect

ronso0 commented:
> maybe you could make the drop-down menu a bit larger

Done, it's much wider now. A while ago I removed the tick mark to make as much as possible of the text fit because there was an issue with the menu, can't recall. Seems to work now, at least on Linux.

Can anyone test this with OSX as well?
If it looks okay there, too, I'd say this ready to merge. Again :)

sblaisot commented:
> Done, it's much wider now.

Much much better !
![image](https://user-images.githubusercontent.com/2768193/32200004-a894a5b0-bdcf-11e7-9e41-8242fbcf0e55.png)

ronso0 commented:
'Effect name eliding' in French looks a bit strange..there's much space left but Effect names are abbreviated..and no tick marks

foss- commented:
@ronso0 have you ever seen covers not being displayed correctly in tango?  https://bugs.launchpad.net/mixxx/+bug/1729451

More OT talk, sry: I applaud all the efforts to improve the skins. I just which, the PRs would be of smaller scale. Fixes could be pushed to master quicker and then be tested and enjoyed by users sooner. I don't see the benefit of having these massive fix collections. There are areas where that makes sense, but for small fixes like the UI glitch above I'd rather see smaller prs being pushed to master more frequently

ronso0 commented:
> have you ever seen covers not being displayed correctly in tango?

Nope, never.
Does it happen in other skins as well?
Does it happen with Covers (not Spinnies)?
Does it happen with small spinnies as well?
Which OS?
Can you reproduce it with a certain file?

> I just wish, PRs would be of smaller scale. Fixes could be pushed to master quicker and then be tested and enjoyed by users sooner.

Yes, you're right.
I claimed this ready to merge quite early. But there was no dev review so far so I kept on fixing small issues and improved even further, other issues popped up during review. As I mentioned earlier this week, this can be merged if there are no issues

ronso0 commented:
@foss- For the screen cast you used Tango from master. In this PR I also optimized the svg files, _maybe_ it's related and _maybe_ this issue is already fixed. Could you test it please? If you need instructions on how to do this quick and easy, let me know.

foss- commented:
I wonder what a good place would be to document steps for that. The [forum > skins announcement section](https://www.mixxx.org/forums/viewforum.php?f=8&sid=f6747b0d0883eda949662c03b470eef0)?
Or better the wiki [skins > skin guidelines](https://www.mixxx.org/wiki/doku.php/skin_guidelines)?

ronso0 commented:
> I wonder what a good place would be to document steps for that. The forum > skins announcement section?

Yes, this is on my ToDo. IMO some links in the wiki need to be placed in a better way, sometimes I fail to find even the simplest stuff and need to resort to search function.
And a sticky thread in the forum is definitely a good place, too!

InannaMoon commented:
So far, the update from 3 days ago is looking great. Thanks.

For what it's worth, the way I have been getting your updates:

1. Click on your name (ronso0) at the top of one of your messages
2. Click on your Mixxx repo
3. Click on Branches
4. Click on Tango update
5. Download a zip file
6. Open zip file
7. Navigate to res/shins directory
8. Open another file browser window
9. Navigate to (Mixxx install location)/skins
10. Rename Tango to TangoN (where is a number)
11. Copy Tango directory from zip browser to skins directory in file browser

Is there an easier way to get to the Tango update branch in your repo?

ronso0 commented:
Cool, thanks. What about the cover art issue on the spinnies?

> Is there an easier way to get to the Tango update branch in your repo?

For a one-time check this probably the easiest way.
If you want to pull this PR every once in a while after changes have been commited, or if you want to fork this PR and propose changes yourself, it's much easier with `git`.
I'm about to create a tutorial for the skin forum.

nikmartin commented:
@InannaMoon, yes, If you are a but sporty with git, PRs in github are just hidden branches in a repo, so you can do this: https://gist.github.com/piscisaureus/3342247 and view all PRs locally in git as branches. Otherwise, you can do a regular merge of a pr: https://help.github.com/articles/checking-out-pull-requests-locally/

sblaisot commented:
> Navigate to (Mixxx install location)/skins
> Rename Tango to TangoN (where is a number)
> Copy Tango directory from zip browser to skins directory in file browser

or under windows to be sure to note have the windows installer system overwrite the file you just dropped (autorepair function), copy the new skin under Tango.new ;)

sblaisot commented:
yesterday I had a 4h set with master + this PR without any usability issue.

LGTM !

Good job @ronso0 and all 👍

ronso0 commented:
@sblaisot Great!
In case your screen is not 1366x768 could you post a screenshot of your ususal configuration?

Commit:
fix knob graphics

sblaisot commented:
The PC I used was not mine this time. I used a PC with a native resolution of 1920x1080 (Full HD) on a 14" screen. That was ... SMALL. But usable. I had to raise library font size a bit.

Unfortunately, I don't have that PC by hand anymore, but here is a screenshot of my main computer with a 22" screen and a resolution of 1680x1050 with the tango skin disposition that best suite my use case.
![image](https://user-images.githubusercontent.com/2768193/32408792-379e1d22-c19f-11e7-9da2-700f6dcb8cea.png)

daschuer commented:
@ronso0 Thank you for all your love putting to this skin. Should I merge it now?

Commit:
fix highlighted FX unit expand button

ronso0 commented:
@daschuer Yes it can be merged!

I fixed two issues:
1. FX unit expand button when an effect was focused before show_focus was set to 0
2. distorted knob pointers
Apparently outlines in svg files were screwed up because [svgo](https://github.com/svg/svgo) "optimized" them too much (with default options). I fixed the graphics and saved them as "Optimized SVG" with Inkscape.
If anyone plans to use svgo -maybe for a build script- this needs to be investigated.

daschuer commented:
Thanks again.

sblaisot commented:
Just an after-merge suggestion for the next tango skin update: add a visual indicator telling if the whole EffectUnit is enabled or disabled. Currently, some mapping allow to disable a whole EffectUnit but we have no visual feedback.

Be-ing commented:
That was intentional. IMO adding an effect chain enable switch overcomplicates the UI. Users get confused with so many switches for different layers and I don't think that was is necessary or very helpful.

sblaisot commented:
ok, so I have to update the Reloop beatmix 2/4 mapping to not use this function as without visual feedback, it is hazardous to disable the whole EffectUnit.
